src/cli.rs

```diff
@@ -619,50 +687,183 @@ fn construct_url(
-    };
-    let mut buf = Vec::new();
-    clap_complete::generate(shell, &mut app, bin_name, &mut buf);
-    let completions = String::from_utf8(buf).unwrap();
-    print!("{}", completions);
+    for &shell in clap_complete::Shell::value_variants() {
+        // Elvish complains about multiple deprecations and these don't seem to work
+        // If you must use them, generate them manually with xh print_completions elvish
+        if shell != clap_complete::Shell::Elvish {
+            clap_complete::generate_to(shell, &mut app, &bin_name, &rest_args[0]).unwrap();
+        }
+    }
     safe_exit();
 }
 
-fn generate_completions(mut app: clap::Command, rest_args: Vec<String>) -> Error {
-    let bin_name = match app.get_bin_name() {
-        Some(name) => name.to_owned(),
-        None => return app.error(ErrorKind::EmptyValue, "Missing binary name"),
-    };
+#[cfg(feature = "man-completion-gen")]
+fn generate_manpages(mut app: clap::Command, rest_args: Vec<String>) -> Error {
+    use chrono::{DateTime, Utc};
+    use roff::{bold, italic, roman, Roff};
+    use std::time::SystemTime;
+
     if rest_args.len() != 1 {
         return app.error(
             ErrorKind::WrongNumberOfValues,
-            "Usage: xh generate_completions <DIRECTORY>",
+            "Usage: xh generate-manpages <DIRECTORY>",
         );
     }
-    for &shell in clap_complete::Shell::value_variants() {
-        // Elvish complains about multiple deprecations and these don't seem to work
-        // If you must use them, generate them manually with xh print_completions elvish
-        if shell != clap_complete::Shell::Elvish {
-            clap_complete::generate_to(shell, &mut app, &bin_name, &rest_args[0]).unwrap();
+
+    let mut roff = Roff::new();
+
+    let items: Vec<_> = app.get_arguments().filter(|i| !i.is_hide_set()).collect();
+    for opt in items.iter().filter(|a| !a.is_positional()) {
+        let mut header = vec![];
+        if let Some(short) = opt.get_short() {
+            header.push(bold(format!("-{}", short)));
+        }
+        if let Some(long) = opt.get_long() {
+            if !header.is_empty() {
+                header.push(roman(", "));
+            }
+            header.push(bold(format!("--{}", long)));
+        }
+        if let Some(value) = &opt.get_value_names() {
+            if opt.get_long().is_some() {
+                header.push(roman("="));
+            } else {
+                header.push(roman(" "));
+            }
+
+            if opt.get_id() == "auth" {
+                header.push(italic("USER"));
+                header.push(roman("["));
+                header.push(italic(":PASS"));
+                header.push(roman("]|"));
+                header.push(italic("TOKEN"));
+            } else {
+                header.push(italic(&value.join(" ")));
+            }
         }
+        let mut body = vec![];
+        if let Some(help) = opt.get_long_help().or_else(|| opt.get_help()) {
```

Comment:
Are there options without help? If not then we might as well panic here

Reply:
Done